src/App.js

```diff
@@ -1,24 +1,41 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useEffect, useState } from "react";
+import Header from "./components/Header/Header";
+import Main from "./pages/Main";
+import firebase from "firebase";
+import { Grid } from "@material-ui/core";
+import { AuthProvider } from "./context/AuthContext";
+import { ModalProvider } from "./context/ModalContext";
+import GeneralModal from "./components/Modal/GeneralModal";
+import { BrowserRouter as Router } from "react-router-dom";
 
 function App() {
+  const [isFireBaseInitialzied, setIsFireBaseInitialzied] = useState(false);
+  useEffect(() => {
+    if (!firebase.apps.length) {
+      firebase.initializeApp({
```

Comment:
Move those variables into a config file, use react environment variables and hide the file from git